### PR TITLE
CRI: add Message in ContainerStatus

### PR DESCRIPTION
**Pull request by yujuhong**
Kubernetes expects a brief CamelCase string as "Reason" and a human-readable
string as "Message" in the conatiner status. It is difficult for kubelet to
derive the other one if the runtime only provides one of the two fields.
Augment the API by adding the "Message" field.

/cc @kubernetes/sig-node

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33386)

### CODE REVIEW

**Comment by yujuhong** [comment]
Note: The alternative is to move the logic of deriving "Reason" back to kubelet, and add fields such as "OOMKilled", etc to CRI. I think that won't work well across runtimes.

/cc @dchen1107 @yifan-gu

**Comment by yifan-gu** [comment]
LGTM

**Comment by yujuhong** [comment]
Fixed the unit tests.

File: pkg/kubelet/dockershim/docker_container.go [hunk 1]
```diff
@@ -261,7 +261,8 @@ func (ds *dockerService) ContainerStatus(containerID string) (*runtimeApi.Contai
 			case r.State.ExitCode == 0:
 				reason = "Completed"
 			default:
-				reason = fmt.Sprintf("Error: %s", r.State.Error)
+				reason = "Error"
+				message = r.State.Error
```
Comment (Random-Liu) [inline pkg/kubelet/dockershim/docker_container.go]: Why only set message here?

File: pkg/kubelet/dockershim/docker_container.go [hunk 2]
```diff
@@ -261,7 +261,8 @@ func (ds *dockerService) ContainerStatus(containerID string) (*runtimeApi.Contai
 			case r.State.ExitCode == 0:
 				reason = "Completed"
 			default:
-				reason = fmt.Sprintf("Error: %s", r.State.Error)
+				reason = "Error"
```
Comment (Random-Liu) [inline pkg/kubelet/dockershim/docker_container.go]: The if-else is wrong to me:
```go
if !finishedAt.IsZero() {
} else if !finishedAt.IsZero() && r.State.ExitCode != 0 { // Doesn't this a subset of !finished.IsZero()
} else {
}
```

**Review by Random-Liu — CHANGES_REQUESTED** [review]

File: pkg/kubelet/dockershim/docker_container.go (hunk 2 above)
Comment (Random-Liu) [inline pkg/kubelet/dockershim/docker_container.go]: This should be:
```go
if !finishedAt.IsZero() {
} else if r.State.ExitCode {
} else {
}
```

Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_container.go]: Oops..done.

**Review by Random-Liu — APPROVED** [review]

**Comment by Random-Liu** [comment]
LGTM

**Pull request merged**